src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs

```diff
@@ -238,6 +239,14 @@ public void GetResult()
             var error = _awaitableError;
             if (error != null)
             {
+                if (error is TaskCanceledException)
```

Comment:
Maybe this could be:

    if (error is InvalidOperationException || error is InvalidOperationException)
    {
       throw error;
    }